### PR TITLE
allow separate grpc listen address

### PR DESCRIPTION
**Pull request by blockloop**

### CODE REVIEW

**Review by bsnyder788 — APPROVED** [review]
LGTM

**Review by thorfour — COMMENTED** [review]
Doesn't this just add a new listener and not route it to the grpc server?

File: web/web.go [hunk 1]
```diff
@@ -416,23 +417,39 @@ func (h *Handler) Reload() <-chan chan error {
...
 	// Monitor incoming connections with conntrack.
-	listener = conntrack.NewListener(listener,
+	webListener = conntrack.NewListener(webListener,
 		conntrack.TrackWithName("http"),
 		conntrack.TrackWithTracing())
 
 	var (
-		m       = cmux.New(listener)
+		m       = cmux.New(webListener)
 		grpcl   = m.Match(cmux.HTTP2HeaderField("content-type", "application/grpc"))
```
Comment (bsnyder788) [inline web/web.go:433]: should we not keep  this one  if they are different?

Comment (blockloop) [reply web/web.go:433]: it's overridden at line 450 by the new one based on the new grpcListener

**Comment by blockloop** [comment]
@bsnyder788 @thorfour please merge

**Comment by thorfour** [comment]
I don't even go here

**Pull request closed without merging**